### PR TITLE
[osmanip] update to v4.6.1

### PR DESCRIPTION
**Pull request by JustWhit3**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
usage:
```
osmanip provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(osmanip CONFIG REQUIRED)
    target_link_libraries(main PRIVATE osmanip::osmanip)
```

When testing usage, the following error occurs:
```
1> [CMake] CMake Error at F:/osmanip/installed/x64-windows/share/osmanip/osmanipTargets.cmake:79 (message):
1> [CMake]   The imported target "osmanip::osmanip" references the file
1> [CMake] 
1> [CMake]      "F:/osmanip/installed/x64-windows/lib/osmanip.lib"
1> [CMake] 
1> [CMake]   but this file does not exist.  Possible reasons include:
1> [CMake] 
1> [CMake]   * The file was deleted, renamed, or moved to another location.
1> [CMake] 
1> [CMake]   * An install or uninstall procedure did not complete successfully.
1> [CMake] 
1> [CMake]   * The installation package was faulty and contained
1> [CMake] 
1> [CMake]      "F:/osmanip/installed/x64-windows/share/osmanip/osmanipTargets.cmake"
1> [CMake] 
1> [CMake]   but not all the files it references.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   F:/osmanip/installed/x64-windows/share/osmanip/osmanipConfig.cmake:11 (include)
1> [CMake]   F:/osmanip/scripts/buildsystems/vcpkg.cmake:855 (_find_package)
1> [CMake]   CMakeUsage/CMakeLists.txt:15 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
```

<details>
<summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "osmanip/utility/strings.hpp"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/osmanip/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")

add_executable (main "CMakeFindUsage.cpp")

find_package(osmanip CONFIG REQUIRED)
target_link_libraries(main PRIVATE osmanip::osmanip)
</pre></details>

**Comment by JustWhit3** [comment]
Hi @MonicaLiu0311 ,

That's very strange. I tried to compile the same program with the same `CMakeLists.txt` file in my computer (which has Ubuntu 22) and it works. Also all the tests succeded. Maybe is something related to your system? I don't know. Did you install the library first, right?

File: ports/osmanip/portfile.cmake [hunk 1]
```diff
@@ -3,23 +3,26 @@ vcpkg_from_github(
...
+    SHA512 17012ebfc06455a933d2acb53cf64ee6b5b452daaed0995b2a0a3e4eee926e819ce7d0d7c83425a484a77fc9919833dbf8d1948d3758c063a89affe1ab83cc82
     HEAD_REF main
 )
 
 # Main commands
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DOSMANIP_TESTS=OFF
+        -DCMAKE_BUILD_TYPE=Release
```
Comment (dg0yt) [inline ports/osmanip/portfile.cmake:14]: NO!

File: ports/osmanip/portfile.cmake [hunk 2]
```diff
@@ -3,23 +3,26 @@ vcpkg_from_github(
...
 vcpkg_cmake_install()
 
 # Move cmake configs
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/osmanip)
 
 # Remove duplicate files
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include"
+                    "${CURRENT_PACKAGES_DIR}/debug/lib" 
+                    "${CURRENT_PACKAGES_DIR}/lib"
```
Comment (dg0yt) [inline ports/osmanip/portfile.cmake:24]: NO! You deleting the libraries here.

File: ports/osmanip/portfile.cmake [hunk 3]
```diff
@@ -3,23 +3,26 @@ vcpkg_from_github(
...
+                    "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (dg0yt) [inline ports/osmanip/portfile.cmake:25]: NO! You are deleting windows debug DLLs here.

File: ports/osmanip/portfile.cmake (hunk 2 above)
Comment (JustWhit3) [reply ports/osmanip/portfile.cmake:24]: Ok i accidentally set up the deletion of the lib dir too, I will fix it now. However shouldn't I delete the content of the debug dir?

File: ports/osmanip/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/osmanip/portfile.cmake:14]: (I just didn't want to start explanations until knowing where to start, but now I got an idea.)
`vcpkg_cmake_configure` sets `CMAKE_BUILD_TYPE`. It builds both `Release` and `Debug`. This shouldn't be touched unless you know what you are doing.

File: ports/osmanip/portfile.cmake (hunk 2 above)
Comment (JustWhit3) [reply ports/osmanip/portfile.cmake]: I removed these lines.

File: ports/osmanip/portfile.cmake (hunk 3 above)
Comment (JustWhit3) [reply ports/osmanip/portfile.cmake]: I removed these lines.

**Comment by dg0yt** [comment]
BTW if you only use Linux, you may get file lists for the other tested triplets from the artifacts in vcpkg CI.

File: ports/osmanip/portfile.cmake [hunk 4]
```diff
@@ -3,23 +3,18 @@ vcpkg_from_github(
...
-    SHA512 cbbae779435bec3995756e1dbc8c283868dd923453054177dfa73bc00ac3510488467e535a1c4b14726f02d198f298eaeb884c1716dcde07e9aec06d02b22d32
+    SHA512 17012ebfc06455a933d2acb53cf64ee6b5b452daaed0995b2a0a3e4eee926e819ce7d0d7c83425a484a77fc9919833dbf8d1948d3758c063a89affe1ab83cc82
     HEAD_REF main
 )
 
 # Main commands
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS
-        -DOSMANIP_TESTS=OFF
```
Comment (dg0yt) [inline ports/osmanip/portfile.cmake:14]: You may need to restore these lines. vcpkg doesn't want to run tests, so we don't need to care for the tests' dependencies.

Comment (JustWhit3) [reply ports/osmanip/portfile.cmake]: Ok I restored the option.

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
osmanip provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(osmanip CONFIG REQUIRED)
    target_link_libraries(main PRIVATE osmanip::osmanip)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**